app/User.php

```diff
@@ -289,4 +321,34 @@ public function scopeInactive($query)
             $q->paid()->current();
         });
     }
+
+    /**
+     * Scope a query to automatically include only access active members
+     * Active: Has paid dues for a currently ongoing term
+     *         or, has a non-zero payment for an active DuesPackage.
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @param mixed $type
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeAccessActive($query)
+    {
+        return $query->whereHas('dues', function ($q) {
+            $q->paid()->accessCurrent();
```

Comment:
This doesn't check override teams

Reply:
That's in the category of SQL I could probably write but really don't want to.

This currently isn't used anywhere except in the access_active attribute getter which already checks teams.